scripts/quests/windurst/Toraimarai_Turmoil.lua

```diff
@@ -0,0 +1,100 @@
+-----------------------------------
+-- Water Way to Go
+-----------------------------------
+-- !addquest 2 16
+-- Ohbiru-Dohbiru : !pos 23 -5 -193 238
+-- Giddeus Spring : !pos -258 -2 -249 145
+-----------------------------------
+require('scripts/globals/interaction/quest')
+require('scripts/globals/items')
+require('scripts/globals/keyitems')
+require('scripts/globals/npc_util')
+require('scripts/globals/quests')
+require('scripts/globals/status')
+require('scripts/globals/zone')
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.WINDURST, xi.quest.id.windurst.TORAIMARAI_TURMOIL)
+
+quest.reward =
+{
+    gil  = 4500,
+    fame = 50,
+    fameArea = xi.quest.fame_area.WINDURST,
+    title = xi.title.CERTIFIED_RHINOSTERY_VENTURER
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+                player:hasCompletedQuest(xi.quest.log_id.WINDURST, xi.quest.id.windurst.BLUE_RIBBON_BLUES) and
+                player:getFameLevel(xi.quest.fame_area.WINDURST) >= 6 and
+                not quest:getMustZone(player)
+        end,
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Ohbiru-Dohbiru'] = quest:progressEvent(785, 4500, xi.keyItem.RHINOSTERY_CERTIFICATE, xi.items.STARMITE_SHELL),
+
+            onEventFinish =
+            {
+                [785] = function(player, csid, option, npc)
+                    if
+                        option == 1
+                    then
+                        quest:begin(player)
+                        npcUtil.giveKeyItem(player, xi.ki.RHINOSTERY_CERTIFICATE)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        --turn in / repeat logic
+        check = function(player, status, vars)
+            return status ~= QUEST_AVAILABLE
+        end,
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Ohbiru-Dohbiru'] =
+            {
+                onTrigger = function(player, npc)
+                    status = player:getQuestStatus(quest.areaId, quest.questId)
```

Comment:
Best to break these out into thier own check headers
Accepted...
Completed....

Reply:
I started that way, but because completed and accepted share the same turn in logic, i didn't want to duplicate code. guess i could have just made a shared function tho